consensus/pbft/viewchange.go

```diff
@@ -435,23 +435,23 @@ func (instance *pbftCore) processNewView() events.Event {
 				continue
 			}
 
-			if _, ok := instance.reqStore[d]; !ok {
-				logger.Warningf("Replica %d missing assigned, non-checkpointed request %s",
+			if _, ok := instance.reqBatchStore[d]; !ok {
+				logger.Warningf("Replica %d missing assigned, non-checkpointed request batch %s",
 					instance.id, d)
-				if _, ok := instance.missingReqs[d]; !ok {
-					logger.Warningf("Replica %v requesting to fetch %s",
+				if _, ok := instance.missingReqBatches[d]; !ok {
+					logger.Warningf("Replica %v requesting to fetch batch %s",
 						instance.id, d)
-					newRequestMissing = true
-					instance.missingReqs[d] = true
+					newReqBatchMissing = true
+					instance.missingReqBatches[d] = true
 				}
 			}
 		}
 	}
 
-	if len(instance.missingReqs) == 0 {
+	if len(instance.missingReqBatches) == 0 {
 		return instance.processNewView2(nv)
-	} else if newRequestMissing {
-		instance.fetchRequests()
+	} else if newReqBatchMissing {
+		instance.fetchRequestBatches()
```

Comment:
But if we add the requests referenced in the `Xset`, as Jason suggests, in the new-view message, then the `fetchRequest` path can be skipped.